Common/SAM.h

```diff
@@ -339,8 +339,7 @@ struct SAMRecord : SAMAlignment {
 			>> o.cigar >> o.mrnm >> o.mpos >> o.isize;
 #if SAM_SEQ_QUAL
 		in >> o.seq >> o.qual;
-		if (in.peek() != '\n')
-			in >> o.tags;
+		getline(in, o.tags);
 #endif
 		in.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
```

Comment:
The tags and EOL character are extracted by `getline(in, o.tags);`, and then this line `in.ignore` will ignore the entire *next* line of the SAM input. The tests passed, which is frightening. Can you please add a unit test with multiple SAM lines and that fails with this code? After verifying that this unit test fails, you can fix this issue.

Reply:
Hmm, the file going in and the file going out has the same number of lines.

Reply:
That would be good news, but I can't reconcile that observation with the code above. `getline` consumes the end-of-line character: `the delimiter character is extracted from input`. https://en.cppreference.com/w/cpp/string/basic_string/getline
The following `in.ignore` would then skip all characters until the following end-of-line character of the next line.

Reply:
Did you test with `SAM_SEQ_QUAL` defined?

Reply:
The sam file has tags in the input and tags in the output so I'm fairly certain `SAM_SEQ_QUAL` is defined.

Reply:
I bet that the caller that you're testing has slurped one single line into an `istringstream`, and you're parsing that single line in your test. The caller is not guaranteed to be using an `istringstream` though. If it used `std::cin` you'd see every other line skipped. You could write a small unit test to test parsing an `istringstream` that contains a few lines of SAM.

Reply:
I see. In any case, can I not just remove line 344 since `getline()` will consume what is remaining until the next new line character?

Reply:
When `SAM_SEQ_QUAL` is defined, yes. When it's not defined, you still need the `.ignore()`.